Functions/Public/Set-GitHubLabel.ps1

```diff
@@ -0,0 +1,108 @@
+    It is possible to add emoji to label names, i.e. 'Good :strawberry:'.
+
+    .PARAMETER Color
+    The new color of the label that is specified by the parameter -Name.
+    The color should be specified without the leading '#'.
+
+   .PARAMETER Description
+    The new description of the label that is specified by the parameter -Name.
+    GitHub has limited the description to max 100 characters.
+
+    .PARAMETER Force
+    Forces the execution without confirmation prompts.
+
+    .EXAMPLE
+    # Update the label name.
+    Set-GitHubLabel -Owner Mary -Repository WebApps -Name 'Label1' -NewName 'NewLabelName'
+
+    .EXAMPLE
+    # Update the label color.
+    Set-GitHubLabel -Owner Mary -Repository WebApps -Name 'good first issue' -Color '5319e7'
+
+    .EXAMPLE
+    # Update the label description.
+    New-GitHubLabel -Owner Mary -Repository WebApps -Name 'good first issue' -Description 'The issue should be easier to fix and can be taken up by a beginner to learn to contribute on GitHub.'
+
+    .EXAMPLE
+    # Update all the label properties at the same time.
+    New-GitHubLabel -Owner Mary -Repository WebApps -Name 'Label1' -NewName 'NewLabelName' -Color '5319e7' -Description 'Label description'
+
+    #>
+    [CmdletBinding(SupportsShouldProcess = $true, ConfirmImpact = 'High')]
+    param (
+        [Parameter(Mandatory = $true, ParameterSetName = 'Repository')]
+        [Alias('User')]
+        [string] $Owner
+      , [Parameter(Mandatory = $true, ParameterSetName = 'Repository')]
+        [string] $Repository
+      , [Parameter(Mandatory = $true, ParameterSetName = 'Repository')]
+        [string] $Name
+      , [Parameter(Mandatory = $false, ParameterSetName = 'Repository')]
+        [string] $NewName
+      , [Parameter(Mandatory = $false, ParameterSetName = 'Repository')]
+        [string] $Color
+      , [Parameter(Mandatory = $false, ParameterSetName = 'Repository')]
+        [string] $Description
+      , [Parameter()]
+        [switch] $Force
+    )
+
+    $shouldProcessCaption = 'Updating an existing GitHub label'
+    $shouldProcessDescription = 'Updating the GitHub label ''{0}'' in the repository ''{1}/{2}''.' -f $Name, $Owner, $Repository
+    $shouldProcessWarning = 'Do you want to update the GitHub label ''{0}'' in the repository ''{1}/{2}''?' -f $Name, $Owner, $Repository
+
+    if ($Force -or $PSCmdlet.ShouldProcess($shouldProcessDescription, $shouldProcessWarning, $shouldProcessCaption)) {
+        $restMethod = 'repos/{0}/{1}/labels/{2}' -f $Owner, $Repository, $Name
+
+        $bodyProperties = @{}
+
+        if ($NewName) {
+            $bodyProperties['name'] = $NewName
+        }
+
+        if ($Color) {
+            $bodyProperties['color'] = $Color
+        }
+
+        if ($Description) {
+            $bodyProperties['description'] = $Description
+        }
+
+        $apiCall = @{
+            Headers =  @{
+                'Accept' = 'application/vnd.github.symmetra-preview+json'
```

Comment:
Style nitpick: don't put quotes around HashTable keys, unless required. In this case, it doesn't appear to be necessary.